internal/api/handlers/categories/handle_delete_category.go

```diff
@@ -18,16 +19,24 @@ func (c *Categories) HandleDeleteCategory(ctx *gin.Context) {
 		return
 	}
 
-	err := c.store.DeleteCategory(ctx, id)
+	category, err := c.store.GetCategory(ctx, id)
 	if err != nil {
-		log.Error().Err(err).Msg("failed to delete category")
-
 		if err == store.ErrNotFound {
 			ctx.Error(error.NewHttpError(http.StatusNotFound))
 		} else {
+			log.Error().Err(err).Msg("failed to get category from store")
 			ctx.Error(error.NewHttpError(http.StatusInternalServerError))
 		}
 
 		return
 	}
+
+	err = c.store.DeleteCategory(ctx, id)
+	if err != nil {
```

Comment:
it looks like code duplication, it will be better to create simple utility for such logic